src/main/java/slimeknights/tconstruct/library/tools/item/ranged/LauncherUserInfo.java

```diff
@@ -0,0 +1,58 @@
+package slimeknights.tconstruct.library.tools.item.ranged;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+
+public interface LauncherUserInfo {
```

Comment:
This interface feels unnecessary; its just a bunch of getters. It could just be parameters for the helper instead.

Reply:
It makes user side easier. Imagine one has a mob that should support both bow and crossbow. By making a record for the mob implementing this interface, the code becomes cleaner and more intuitive.

Reply:
Actually I feel like direction should be part of LauncherUserInfo as well: it's viewVector for player, and some calculated projection vector for mobs, based on target position

Reply:
> It makes user side easier. Imagine one has a mob that should support both bow and crossbow. By making a record for the mob implementing this interface, the code becomes cleaner and more intuitive.

I disagree. If your mob does both, just make two helper methods that call our method. We do it all the time in our API. And its just 4 values. Make a method for players to call that automatically sets the properties, and make a general method that other mods can call, making their own helper if they wish.

Reply:
I will change it then.